### PR TITLE
Adds docs for folder_watcher component

### PR DESCRIPTION
**Pull request by robmarkcole**
**Description:**
Adds documentation for the new folder_watcher component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant/pull/12918) (if applicable):** home-assistant/home-assistant#12918

## Checklist:

- [ x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [ x] The documentation follow the [standards][standards].

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
@robmarkcole Looks great! Thanks!
Can be merged as soon as the parent PR gets merged 🥇

**Comment by robmarkcole** [comment]
Cheers @frenck

**Comment by frenck** [comment]
Rebased the PR onto `next`

**Review by frenck — DISMISSED** [review]

**Comment by frenck** [comment]
@robmarkcole Finally managed to get the build to succeed 😠 
So, now its truly ready to merge... (as soon as the parent PR gets merged that is...)

🎉

**Comment by robmarkcole** [comment]
Cheers @frenck

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Can be merged as soon as the build passes.

File: source/_components/folder_watcher.markdown [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  default: "*" 
+  type: string
+{% endconfiguration %}
+
+## Patterns
+
+Pattern matching using [fnmatch](https://docs.python.org/3.6/library/fnmatch.html) can be used to limit filesystem monitoring to only files which match the configured patterns. The following example shows the configuration required to only monitor filetypes `.yaml` and `.txt`.
+
+```yaml
+{% raw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:45]: This needs to be moved to *outside* of the code block.

File: source/_components/folder_watcher.markdown [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+folder_watcher:
+  watchers:
+    - folder: /config
+      patterns:
+        - '*.yaml'
+        - '*.txt'
+{% raw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:52]: This needs to be moved outside of the code block, and changed to `{% endraw %}`

File: source/_components/folder_watcher.markdown [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+```
+
+## Automations
+
+Automations can be triggered on filesystem event data using a data_template. The following automation will send a notification with the name and folder of new files added to that folder:
+
+```yaml
+{% raw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:60]: Move to outside of code block.

File: source/_components/folder_watcher.markdown [hunk 4]
```diff
@@ -0,0 +1,77 @@
+---
+layout: page
+title: "folder watcher"
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:3]: This should be cased properly as `Folder Watcher`

File: source/_components/folder_watcher.markdown [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    - folder: /config
+      patterns:
+        - '*.yaml'
+        - '*.txt'
+{% raw %}
+```
+
+## Automations
+
+Automations can be triggered on filesystem event data using a data_template. The following automation will send a notification with the name and folder of new files added to that folder:
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:57]: Please wrap **data_template** with backticks to make it a code block.

File: source/_components/folder_watcher.markdown [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        file: "{{trigger.event.data.path}}"
+    service: notify.pushbullet
+  alias: New file alert
+  condition: []
+  id: '1520092824697'
+  trigger:
+  - event_data: {"event_type":"created"}
+    event_type: folder_watcher
+    platform: event
+{% endraw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:75]: Move to outside of code block.

File: source/_components/folder_watcher.markdown [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+## Automations
+
+Automations can be triggered on filesystem event data using a data_template. The following automation will send a notification with the name and folder of new files added to that folder:
+
+```yaml
+{% raw %}
+- action:
+  - data_template:
+      message: 'Created {{trigger.event.data.file}} in {{trigger.event.data.folder}}'
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:63]: Please add one space after the Jinja opening delimiters (`{{`), and one space before the Jinja closing delimiters (`}}`).

File: source/_components/folder_watcher.markdown [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+      title: New image captured!
+      data:
+        file: "{{trigger.event.data.path}}"
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:66]: Please add one space after the Jinja opening delimiter (`{{`), and one space before the Jinja closing delimiter (`}}`).

File: source/_components/folder_watcher.markdown [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: System Monitor
+ha_iot_class: "Local Polling"
+ha_release: 0.67
+---
+
+This component adds [Watchdog](https://pythonhosted.org/watchdog/) file system monitoring, publishing events on the Home-Assistant bus on the creation/deletion/modification of files.
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:16]: Please change **Home-Assistant** to **Home Assistant**.

File: source/_components/folder_watcher.markdown [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+To configure the `folder_watcher` component add to you `configuration.yaml` file:
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:18]: Change to:

"To enable the Folder Watcher component in your installation, add the following to your `configuration.yaml` file:"

File: source/_components/folder_watcher.markdown [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+
+```yaml
+{% raw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:21]: Move to outside of code block.

File: source/_components/folder_watcher.markdown [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+folder_watcher:
+  watchers:
+    - folder: /config
+{% endraw %}
```
Comment (dale3h) [inline source/_components/folder_watcher.markdown:25]: Move to outside of code block.

**Review by dale3h — COMMENTED** [review]
There are a couple of minor issues that need to be changed on this docs page before we launch 0.67.

File: source/_components/folder_watcher.markdown (hunk 1 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:45]: ok

File: source/_components/folder_watcher.markdown (hunk 2 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:52]: ok

File: source/_components/folder_watcher.markdown (hunk 3 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:60]: ok

File: source/_components/folder_watcher.markdown (hunk 4 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:3]: ok

File: source/_components/folder_watcher.markdown (hunk 5 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:57]: ok

File: source/_components/folder_watcher.markdown (hunk 6 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:75]: ok

File: source/_components/folder_watcher.markdown (hunk 7 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:63]: ok

File: source/_components/folder_watcher.markdown (hunk 8 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:66]: ok

File: source/_components/folder_watcher.markdown (hunk 9 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:16]: ok

File: source/_components/folder_watcher.markdown (hunk 10 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:18]: ok

File: source/_components/folder_watcher.markdown (hunk 11 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:21]: ok

File: source/_components/folder_watcher.markdown (hunk 12 above)
Comment (robmarkcole) [reply source/_components/folder_watcher.markdown:25]: ok

**Pull request merged**